### ISSUE TITLE
Document play.http.context configuration

### ISSUE THREAD

**Issue by richdougherty** [playframework/playframework#4084]
Play framework supports adding a base URL to the root of the application path. This is configured with the `play.http.context` configuration setting in Play 2.4. (Previously it was the `application.context` setting.)

We need to check the setting's behaviour and then document it. To start working out its behaviour, look at the `RoutesProvider`, the `HttpConfiguration.fromConfiguration` method and the generated reverse router code's logic.

**Comment by richdougherty** [comment]
Cc @huntc @markusjura.

**Comment by wsargent** [comment]
This is covered by the deprecated setting and by https://www.playframework.com/documentation/2.4.x/Migration24#Configuration-changes

```
val config = PlayConfig(configuration)
    val context = {
      val ctx = config.getDeprecated[String]("play.http.context", "application.context")
      if (!ctx.startsWith("/")) {
        throw configuration.globalError("play.http.context must start with a /")
      }
      ctx
    }
```

**Comment by nafg** [comment]
The setting is not supported anymore?

**Comment by mkurz** [comment]
It is still supported via `play.http.context`.
The conf key `application.context` is deprecated.

**Comment by nafg** [comment]
So it's not documented except in the migration guide? What if someone is
starting with play 2.4+?

**Comment by gmethvin** [comment]
I agree with @nafg. We have a few things that only appear to be documented in migration guides, such as https://www.playframework.com/documentation/2.4.x/PluginsToModules. Those should show up somewhere in the normal documentation. At least we could embed the relevant reference.conf in the page, which has those settings.

**Comment by wsargent** [comment]
If you're starting with Play 2.4, then it runs "application.context".  If you move to 2.5, then running will say that "application.context" is deprecated and tell you the new setting, "play.http.context".

We can add it as a commented out setting in https://github.com/playframework/playframework/blob/master/templates/play-common/conf/application.conf#L85 if that helps...

**Comment by thomhickey** [comment]
Yes please ... I just wasted a bunch of time (ok not really because it's all learning I suppose) looking for a way to have my app live behind a proxy (myhost.com/my_app/) and it turned out to be really simple (play.http.context).  Surprised not to find a reference that documents every possible config property.
